akka-cluster/src/main/scala/akka/cluster/ClusterHeartbeat.scala

```diff
@@ -249,14 +269,35 @@ private[cluster] case class HeartbeatNodeRing(selfAddress: UniqueAddress, nodes:
   /**
    * The receivers to use from a specified sender.
    */
-  def receivers(sender: UniqueAddress): immutable.Set[UniqueAddress] =
+  def receivers(sender: UniqueAddress): Set[UniqueAddress] =
     if (useAllAsReceivers)
       nodeRing - sender
     else {
-      val slice = nodeRing.from(sender).tail.take(monitoredByNrOfMembers)
-      if (slice.size < monitoredByNrOfMembers)
-        (slice ++ nodeRing.take(monitoredByNrOfMembers - slice.size))
-      else slice
+
+      // Pick nodes from the iterator until n nodes that are not unreachable have been selected.
+      // Intermediate unreachable nodes are also included in the result.
```

Comment:
It is actually impossible since we just walk around the node ring, and the “last” node of the live cluster will then monitor all its unreachable neighbors on the ring.